src/core/Akka/IO/TlsConnection.cs

```diff
@@ -0,0 +1,1183 @@
+                Handler = handler;
+                KeepOpenOnPeerClosed = keepOpenOnPeerClosed;
+                UseResumeWriting = useResumeWriting;
+            }
+        }
+
+        // INTERNAL MESSAGES
+        private sealed class UpdatePendingWriteAndThen : INoSerializationVerificationNeeded
+        {
+            public Option<PendingWrite> RemainingWrite { get; }
+            public Action Work { get; }
+
+            public UpdatePendingWriteAndThen(Option<PendingWrite> remainingWrite, Action work)
+            {
+                RemainingWrite = remainingWrite;
+                Work = work;
+            }
+        }
+
+        private sealed class WriteFileFailed
+        {
+            public SocketException Cause { get; }
+
+            public WriteFileFailed(SocketException cause)
+            {
+                Cause = cause;
+            }
+        }
+
+        private abstract class PendingWrite
+        {
+            public IImmutableList<(IActorRef Commander, object Ack)> PendingAcks { get; }
+
+            protected PendingWrite(IImmutableList<(IActorRef Commander, object Ack)> pendingAcks)
+            {
+                PendingAcks = pendingAcks;
+            }
+
+            public abstract Task DoWrite(ConnectionInfo info);
+        }
+
+        private sealed class PendingBufferWrite : PendingWrite
+        {
+            private readonly TlsConnection _connection;
+            private readonly IActorRef _self;
+            private readonly IEnumerable<ByteString> _dataToSend;
+            private readonly IBufferPool _bufferPool;
+            private readonly SendReceiveArgsFake _sendArgs;
+
+            public PendingBufferWrite(
+                TlsConnection connection,
+                SendReceiveArgsFake sendArgs,
+                IActorRef self,
+                IImmutableList<(IActorRef Commander, object Ack)> acks,
+                IEnumerable<ByteString> dataToSend,
+                IBufferPool bufferPool) : base(acks)
+            {
+                _connection = connection;
+                _sendArgs = sendArgs;
+                _self = self;
+                _dataToSend = dataToSend;
+                _bufferPool = bufferPool;
+            }
+
+            public override async Task DoWrite(ConnectionInfo info)
+            {
+                try
+                {
+                    //_sendArgs.SetBuffer(_dataToSend);
+#pragma warning disable 4014
+                    Task.Run(async () =>
+                    {
+                        foreach (var byteString in _dataToSend)
+                        {
+                            foreach (var buffer in byteString.Buffers)
+                            {
+                                await _connection.SslStream.WriteAsync(
+                                    buffer.Array,
+                                    buffer.Offset, buffer.Count);
+                            }
+                        }
+
+                        return NotUsed.Instance;
```

Comment:
This is probably Bad and Wrong. There's definitely some ugly going on:
  - Task.Run is a smell
  - Lots of back-and-forth here with awaits when we have small buffers (i.e. after manual serialization/framing/batching)